File: api/src/main/java/me/devsnox/planetsystem/api/holder/data/PlayerData.java
package me.devsnox.planetsystem.api.holder.data;

import me.devsnox.planetsystem.api.player.PlanetPlayer;
import org.jetbrains.annotations.Nullable;

import java.util.Set;
import java.util.UUID;
import java.util.function.Consumer;

public interface PlayerData {

    void load(UUID uuid, Consumer<PlanetPlayer> request);

    void save(UUID uuid);

    void unload(UUID uuid);

    @Nullable
    default PlanetPlayer getPlayer(final UUID uuid) {
        for (final PlanetPlayer player : this.getPlayers()) if (player.getUUID().equals(uuid)) return player;
        return null;
    }

    Set<PlanetPlayer> getPlayers();

    //Planet getPlanet(UUID owner);

    //LoadedPlanet getLoadedPlanetByOwner(UUID owner);

}

File: api/src/main/kotlin/me/devsnox/planetsystem/api/holder/data/PlayerData.kt
package me.devsnox.planetsystem.api.holder.data

import me.devsnox.planetsystem.api.player.PlanetPlayer
import java.util.*

interface PlayerData {

	val players: Set<PlanetPlayer>

	fun load(uuid: UUID, request: (PlanetPlayer) -> Unit)

	fun save(uuid: UUID)

	fun unload(uuid: UUID)

	fun getPlayer(uuid: UUID): PlanetPlayer? = players.find { it.uuid == uuid }

}
